### DISCUSSION TITLE
Security Question

### DISCUSSION THREAD

**Discussion by tvarga94** [austintoddj/canvas#1436] [Troubleshooting]
Hello I have a website that uses Canvas for blogging (the website is still under development), but since I activated the ssl I see strange behavior. 

First, I see that notice/error:
![Screenshot from 2024-09-08 21-59-18](https://github.com/user-attachments/assets/31f8fd1a-a7f0-4944-9cfb-8faf60cd1130)

Second, which is more concerning, when I login to the site, I get logged out right after...Did someone else experience that? and is it something that canvas does or maybe a server problem?

Thank you for the answers, it would be great to discuss this as soon as possible, we are about to go live
Br, Tamas

@austintoddj

**Comment by tvarga94** [comment]
When I turn off my firewall on the server and https (no ssl) it works, but as soon as I turn them back on canvas does not work properly.

Is there anyone who runs canvas with ssl, firewall and https on? It would be great to know that the canvas code is not to blame here

**Comment by austintoddj** [comment]
Closing this as a host/session configuration issue rather than a Canvas product bug.

In v7, Canvas no longer owns login or sessions — authentication is entirely the host app’s responsibility (`CANVAS_GUARD`). Mixed HTTP/HTTPS, cookie `Secure`/`SameSite`, or domain mismatches after enabling SSL almost always come from the Laravel/session/proxy setup, not Canvas routes.

If you still hit logout-after-login on a clean [v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0) install with a correct `APP_URL` and session config, please open a new issue with host details.
